cachi2/interface/cli.py

```diff
@@ -408,16 +428,33 @@ def merge_sboms(
     first to produce SBOMs to merge.
     """
     sboms_to_merge = []
+    if sbom_type == SBOMFormat.cyclonedx:
+        sbom_cls: Union[Type[Sbom], Type[SPDXSbom]] = Sbom
+    else:
+        sbom_cls = SPDXSbom
     for sbom_file in sbom_files_to_merge:
         try:
-            sboms_to_merge.append(Sbom.model_validate_json(sbom_file.read_text()))
-        except pydantic.ValidationError:
+            sboms_to_merge.append(sbom_cls.model_validate_json(sbom_file.read_text()))
+        except pydantic.ValidationError as e:
+            print(e)
             raise UnexpectedFormat(f"{sbom_file} does not appear to be a valid Cachi2 SBOM.")
-    sbom = Sbom(
-        components=merge_component_properties(
-            chain.from_iterable(s.components for s in sboms_to_merge)
+    if sbom_type == SBOMFormat.cyclonedx:
+        sbom: Union[Sbom, SPDXSbom] = Sbom(
+            components=merge_component_properties(
+                chain.from_iterable(cast(Sbom, s).components for s in sboms_to_merge)
+            )
         )
-    )
+    else:
+        packages = chain.from_iterable(cast(SPDXSbom, s).packages for s in sboms_to_merge)
+        sbom = SPDXSbom(
+            spdxVersion="SPDX-2.3",
+            spdxIdentifier="SPDXRef-DOCUMENT",
+            dataLicense="CC0-1.0",
+            name=cast(SPDXSbom, sboms_to_merge[0]).name,
```

Comment:
It looks like ordering of an input will affect name and creationInfo of output. Is this really desired?

Reply:
well, how else would you handle it? As it's meant to merge only cachi2 produced sboms, creation info should be same for all. And name needs to be take from one of those